lib/couchbase/protostellar/request_generator/search.rb

```diff
@@ -62,6 +65,11 @@ def search_query_request(index_name, query, options)
             **proto_opts
           )
 
+          # Set the search facets in the request
+          get_facets(options).each do |key, facet|
+            proto_req.facets[key] = facet
+          end
```

Comment:
This is somewhat ugly but I've not figured out how to initialize a [Protouf Map](https://protobuf.dev/reference/ruby/ruby-generated/#map-fields) where the values are a custom type (like `Facet` here), so I'm just filling in the one that's created when the request is created.

Reply:
I think it looks good.

The only option I managed to find is the following

```suggestion
          # Set the search facets in the request
          proto_req.facets = proto_req.facets.merge(get_facets(options))

```

 but I doubt it is better, as it will create another copy of the Protobuf::Map, as it does not have `#update` or `#merge!` to do it in place

```
irb(main):039:0> Couchbase::Protostellar::Generated::Search::V1::SearchQueryRequest.new.facets.public_methods(false)
=> [:dup, :delete, :clear, :==, :freeze, :inspect, :clone, :[], :[]=, :keys, :values, :length, :size, :merge, :has_key?, :each, :to_h, :hash]
irb(main):040:0> Couchbase::Protostellar::Generated::Search::V1::SearchQueryRequest.new.facets.respond_to? :update
=> false
```